lib/Parse/ParseIfConfig.cpp

```diff
@@ -407,7 +407,19 @@ class EvaluateIfConfigCondition :
 
   bool visitUnresolvedDeclRefExpr(UnresolvedDeclRefExpr *E) {
     auto Name = getDeclRefStr(E);
-    return Ctx.LangOpts.isCustomConditionalCompilationFlagSet(Name);
+
+    // Check whether this is any one of the known compiler features.
+    const auto &langOpts = Ctx.LangOpts;
+    bool isKnownFeature = llvm::StringSwitch<bool>(Name)
+#define LANGUAGE_FEATURE(FeatureName, SENumber, Description, Option) \
+        .Case("$" #FeatureName, Option)
```

Comment:
I would recommend using some prefix in addition to the `$`, like `"$has" #FeatureName`, so that if we find other things we'd like to use `$` defines to test for, we can give them a separate prefix so they don't collide with language features.